File: source_funcs/go_transcoder/FIND_THE_LARGEST_SUBARRAY_WITH_0_SUM/FIND_THE_LARGEST_SUBARRAY_WITH_0_SUM.go
package main

func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}

func f_gold(arr []int, n int) int {
	var max_len int = 0
	for i := int(0); i < n; i++ {
		var curr_sum int = 0
		for j := int(i); j < n; j++ {
			curr_sum += arr[j]
			if curr_sum == 0 {
				max_len = max(max_len, j-i+1)
			}
		}
	}
	return max_len
}

File: rust_funcs/go_transcoder/FIND_THE_LARGEST_SUBARRAY_WITH_0_SUM/FIND_THE_LARGEST_SUBARRAY_WITH_0_SUM.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}



fn f_gold(arr: [i32;2], n: i32) -> i32 {
    let mut max_len = 0;
    for i in 0..n {
        let mut curr_sum = 0;
        for j in i..n {
            curr_sum += arr[j as usize];
            if curr_sum == 0 {
                max_len = max(max_len, (j - i + 1) as i32);
            }
        }
    }
    max_len
}
